python/test/unit/language/test_core.py

```diff
@@ -2545,7 +2545,20 @@ def kernel(X, Y, N, BLOCK_M: tl.constexpr, BLOCK_N: tl.constexpr, NUM_PID_N: tl.
 @pytest.mark.parametrize("M, N", [[32, 16], [32, 32], [32, 64], [64, 32]])
 @pytest.mark.parametrize("src_layout", scan_layouts)
 @pytest.mark.parametrize("axis", [0, 1])
-def test_scan_layouts(M, N, src_layout, axis, device, tmp_path: pathlib.Path):
+@pytest.mark.parametrize("add_overflow_check", [False, True])
+def test_scan_layouts(M, N, src_layout, axis, add_overflow_check, device, tmp_path: pathlib.Path):
+
+    overflow_check = """
+        %17 = arith.extsi %arg2 : i32 to i64
+        %18 = arith.extsi %arg3 : i32 to i64
+        %19 = arith.addi %17, %18 : i64
+        %i32.min = arith.constant -2147483648: i64
+        %i32.max = arith.constant 2147483647: i64
+        %20 = arith.cmpi slt, %19, %i32.max : i64
+        %21 = arith.cmpi sge, %19, %i32.min : i64
+        %22 = arith.andi %20, %21 : i1
+        tt.assert %22, "overflow detected" : i1
+    """
```

Comment:
isn't this likely to break often whenever we do small changes? Should this be a lit test instead?

Reply:
I want to check that the code produces the expected result, so I don't think a lit test helps here. Also not sure what changes you had in mind, this would only break if there are significant changes to the ir format which seems unlikely.

Reply:
if a pass slightly change the IR the value name may be different for instance? In general that's why in LIT tests we try not to hardcode value names. 

> I want to check that the code produces the expected result, so I don't think a lit test helps here.

ah right

Anyway not a blocker but we may have to find something better if it does break when changing unrelated part of the code

Reply:
> if a pass slightly change the IR the value name may be different for instance? 

We're providing the gpu ir directly here so there are no passes applied to this IR. The only way the argument names might change is if you manually edit the test itself.